server: remove createTestNode

tbg opened the pull request:
This crufty method got in the way of improving the server startup sequence. Luckily, it was easy to remove it.

tbg committed:
server: simplify TestCorruptedClusterID

Now it has no need to start up a Node, and the check should have
moved to inspectEngines a long time ago anyway.

Release note: None

tbg committed:
server: simplify TestBootstrapNewStore

Avoids yet another caller of `createTestNode`, which I want to phase
out.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46841)
<!-- Reviewable:end -->

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46841#-:-M3q9yLwCgHhuErZVJSp:bb6ywmx)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

___
*[pkg/server/node_test.go, line 154 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46841#-M3q9vWR2UOT872IXNIZ:-M3q9vWR2UOT872IXNI_:b4qay90) ([raw file](https://github.com/cockroachdb/cockroach/blob/59b5882577d298cdeed6a1d9c582439551a1e7ab/pkg/server/node_test.go#L154)):*
> ```Go
> 
> // createAndStartTestNode creates a new test node and starts it. The server and node are returned.
> func createAndStartTestNode(
> ```

good riddance
___
*[pkg/server/node_test.go, line 290 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46841#-M3q9Cbn5-eUdPfT24Pn:-M3q9Cbn5-eUdPfT24Po:b-8mzl9h) ([raw file](https://github.com/cockroachdb/cockroach/blob/59b5882577d298cdeed6a1d9c582439551a1e7ab/pkg/server/node_test.go#L290)):*
> ```Go
> 
> 	// Check whether all stores are started properly.
> 	testutils.SucceedsSoon(t, func() error {
> ```

nit: if you're interested, I'd take the opportunity to introduce a way to signal from `StartServer` when the bootstrapping is done, so we can get rid of the `SucceedsSoon`. Some testing knob channel.
Otherwise, keep that comment from before about the goroutine. Which I find dubious, actually. How come we can otherwise use the server immediately is the bootstrap is async?
___
*[pkg/server/server.go, line 1000 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46841#-M3q8nbf4q2K-vHhvdNE:-M3q8nbf4q2K-vHhvdNF:b-ifgw07) ([raw file](https://github.com/cockroachdb/cockroach/blob/c6cbb2106cfc8ac9504ee376f199595c7d645d91/pkg/server/server.go#L1000)):*
> ```Go
> 		if storeIdent.StoreID == 0 || storeIdent.NodeID == 0 || storeIdent.ClusterID == uuid.Nil {
> 			return nil, nil, clusterversion.ClusterVersion{},
> 				errors.Errorf("partially initialized store: %+v", storeIdent)
> ```

nit: `AssertionErrorf`?


<!-- Sent from Reviewable.io -->

tbg committed:
server: simplify TestNodeJoin, TestStartNodeWithLocality

We now get to delete `createTestNode`, which was ultimately prompted
this work, as this method got in the way of improving the server
startup sequence.

Release note: None

Release justification:

tbg reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46841#-:-M3qCuHB7hhk_CpMstWJ:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

___
*[pkg/server/node_test.go, line 290 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46841#-M3q9Cbn5-eUdPfT24Pn:-M3qBpn9BhxV_kGTI8De:bcioink) ([raw file](https://github.com/cockroachdb/cockroach/blob/59b5882577d298cdeed6a1d9c582439551a1e7ab/pkg/server/node_test.go#L290)):*
> Which I find dubious, actually. How come we can otherwise use the server immediately is the bootstrap is async?

It's subtle. Right now the SucceedsSoon isn't needed (we init all stores sychronously), but the moment we fix https://github.com/cockroachdb/cockroach/issues/39415 it may be needed again. So instead of setting myself a trap, I will leave as is. BTW, this is not about bootstrapping. It's about initializing any extra stores that we have (on top of at least one that's already initialized). Bootstrap is always done by the time the server startup returns.

I'm not interested in changing more in this PR. Plenty of them will be coming your way. I can promise more good riddances.
___
*[pkg/server/server.go, line 1000 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46841#-M3q8nbf4q2K-vHhvdNE:-M3qCQbh8V3YgLwxQzrf:bq88xbr) ([raw file](https://github.com/cockroachdb/cockroach/blob/c6cbb2106cfc8ac9504ee376f199595c7d645d91/pkg/server/server.go#L1000)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

nit: `AssertionErrorf`?
</blockquote></details>

Any error return in much of this file is really expected to never fire and comes out of exactly one place. I think we should use the decorated assertions only in places where they add value. Or maybe not - but just changing this one random line isn't an option and if we're going to change them all, then not here.


<!-- Sent from Reviewable.io -->

tbg commented:
bors r=andreimatei
